atom/browser/api/atom_api_web_contents.cc

```diff
@@ -1936,6 +1944,26 @@ void WebContents::OnGetZoomLevel(content::RenderFrameHost* rfh,
   rfh->Send(reply_msg);
 }
 
+void WebContents::OnGetPreloadScript(content::RenderFrameHost* rfh,
+                                     IPC::Message* reply_msg) {
+  std::string preload_script;
+
+  auto* web_preferences = WebContentsPreferences::From(web_contents());
+  if (web_preferences) {
+    base::FilePath::StringType preload;
+    if (web_preferences->dict()->GetString(options::kPreloadScript, &preload)) {
+      base::ScopedAllowBlockingForTesting allow_blocking;
```

Comment:
calling fs.readFileSync would block the main process before as well. in this case it would be possible to load the file on a different thread and send the reply when it's done.